File: src/main/scala/com/ajjpj/amapper/AMapper.scala
package com.ajjpj.amapper

import com.ajjpj.amapper.core.{ADiff, AQualifier, AType}

/**
 * @author arno
 */
trait AMapper {
  def map(source: AnyRef, sourceType: AType, sourceQualifier: AQualifier, target: AnyRef, targetType: AType, targetQualifier: AQualifier): Option[AnyRef]
  def diff(sourceOld: AnyRef, sourceNew: AnyRef, sourceType: AType, sourceQualifier: AQualifier, targetType: AType, targetQualifier: AQualifier): ADiff
}

File: src/main/java/com/ajjpj/amapper/AMapper.java
package com.ajjpj.amapper;

import com.ajjpj.amapper.core.diff.ADiff;
import com.ajjpj.amapper.core.tpe.AQualifier;
import com.ajjpj.amapper.core.tpe.AType;
import com.ajjpj.amapper.util.coll.AOption;


/**
 * @author arno
 */
public interface AMapper {
    AOption<Object> map(Object source, AType sourceType, AQualifier sourceQualifier, Object target, AType targetType, AQualifier targetQualifier);
    ADiff diff(Object sourceOld, Object sourceNew, AType sourceType, AQualifier sourceQualifier, AType targetType, AQualifier targetQualifier);
}
